persistence/CloudProxyWrapper.js

```diff
@@ -9,13 +12,31 @@ class CloudProxyWrapper {
     connect() {
     }
 
+    localToRemote(path) {
+        if (path.startsWith(objectsPath)) {
+            let relPath = pathOps.relative(objectsPath, path);
+            if (!relPath.startsWith('..')) {
```

Comment:
I believe `..` can be within a path, e.g. `folder/other/../file.ext`

Reply:
In theory `pathOps.relative` should normalize `path` to remove that sort of thing but that isn't explicit in its documentation so I'll get more paranoid with it

Reply:
Oh I see, I was a bit confused by checking for `startsWith('..')` if relative was meant to remove that, but I assume it can start with `..` if it's in a higher relative directory. Looks good to me as is now that I understand how that interaction works.

Reply:
Adjusted this function and remoteToLocal to be a bit stronger and clearer in their guarantees